daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -283,10 +283,15 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
       seeding: speedy.InitialSeeding,
   )(implicit loggingContext: LoggingContext): Result[(SubmittedTransaction, Tx.Metadata)] =
     for {
-      sexpr <- runCompilerSafely(
+      disclosureUpdates <- runCompilerSafely(
+        NameOf.qualifiedNameOfCurrentFunc,
+        compiledPackages.compiler.unsafeCompileContractDisclosures(disclosures),
+      )
+      commandEvals <- runCompilerSafely(
         NameOf.qualifiedNameOfCurrentFunc,
         compiledPackages.compiler.unsafeCompile(commands),
       )
+      sexpr = SELet1General(disclosureUpdates, commandEvals) // FIXME: is this correct?
```

Comment:
I  think it would be helpful have an entry point in `Compiler.scala`, (perhaps `unsafeCompileWithDisclosures`) which constructs the speedy let expression, and so we only make one call here to `runCompilerSafely` here.